src/helpers/augmentedReality/multipleSceneIndexGenerator.ts

```diff
@@ -47,14 +73,56 @@ export const multipleSceneIndexGenerator = ({
     const texture = Math.floor(currentDayNumber / timePeriodInDays);
     const model = Math.floor(texture / variableTexturesCount);
     modelIndex = model % scenesCount;
-    const textureIndex = texture % variableTexturesCount;
-    const variableTexture = variableTextures?.[textureIndex];
+
+    // When `variableTextures` are created, they are created according to index 0 of the scene array.
+    // Once the `modelIndex` is found, it must be updated to select the correct texture file.
+    if (modelIndex > 0) {
+      variableTextures = scenes[modelIndex]?.localUris?.filter(
+        ({ type, stable }) => type === 'texture' && !stable
+      );
+    }
+
     localUris = scenes?.[modelIndex]?.localUris;
+    const textureIndex = texture % variableTexturesCount;
+    textures.push(variableTextures?.[textureIndex] as TTextures[0]);
 
-    _remove(localUris, ({ type, stable }) => type === 'texture' && !stable);
+    localUris.forEach((subItem) => {
+      modelsAndTexturesArrayGenerator(subItem, models, textures, !!subItem.stable);
+    });
 
-    !!variableTexture && localUris.push(variableTexture);
+    // deleted to avoid sending unused information to the `AugmentedRealityView` screen
+    _remove(localUris, ({ type }) => type === 'texture' || type === 'vrx');
+
+    return { localUris, models, textures };
+  }
+
+  for (let i = 0; i < scenes.length; i++) {
+    localUris = scenes?.[i]?.localUris;
+    localUris.forEach((subItem) => {
+      modelsAndTexturesArrayGenerator(subItem, models, textures);
+    });
   }
 
-  return { localUris };
+  // deleted to avoid sending unused information to the `AugmentedRealityView` screen
+  _remove(localUris, ({ type }) => type === 'texture' || type === 'vrx');
+
+  return { localUris, models, textures };
+};
+
+const modelsAndTexturesArrayGenerator = (
+  subItem: TScenes['localUris'][0],
+  models: TModels,
+  textures: TTextures,
+  shouldPushTexture: boolean = true
```

Comment:
![image](https://user-images.githubusercontent.com/1942953/206694092-c180c8ab-954a-43aa-9be9-1a8bb4533050.png)

`shouldPushTexture = true` should  be enough

Reply:
👍
https://github.com/ikuseiGmbH/smart-village-app-app/commit/d7995b7f7ee9213263362f5fdb7ab6b04cf89d0d